### PR TITLE
Parallel ingestion

### LINKED ISSUE #2677: Research parallel domain ingestion across multiple connections
### Problem

Currently domain ingestion in the importer is performed sequentially.
This prevents scaling and doesn't capitalize on the fact that not all tables require reconciliation during inserts

### Solution

Concurrently insert into tables across multiple connections

- Group table inserts 
- Within groups allow for parallel ingestion execution

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by Nana-EC**
**Description**:
Currently domain ingestion in the importer is performed sequentially.
This prevents scaling and doesn't capitalize on the fact that not all tables require reconciliation during inserts

- Add a batch processing flow that runs PgCopy/UpsertPgCopy tasks in parallel where applicable
- Add configuration to turn parallel processing on and modify ExecutorService thread pool
- Update confirmation.md file with configurations
- Update performance tests 

**Related issue(s)**:

Fixes #2677 

**Notes for reviewer**:
Still investigating failing `RecordFileParserIntegrationTest` tests. They pass locally but for some reason in CI aren't picking up some of the recordItems

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/domain/DomainBuilder.java [hunk 1]
```diff
@@ -79,4 +84,48 @@ private long id() {
...
+                .decimals(1000)
+                .freezeDefault(false)
+                .freezeKey(hexKey)
+                .initialSupply(1_000_000L)
+                .kycKey(hexKey)
+                .maxSupply(1_000_000_000L)
+                .modifiedTimestamp(timestamp)
+                .name("FOO COIN TOKEN")
+                .pauseKey(hexKey)
+                .pauseStatus(TokenPauseStatusEnum.NOT_APPLICABLE)
```
Comment (ijungmann) [inline hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/domain/DomainBuilder.java:119]: Should either not have pause key or be `UNPAUSED`

File: hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/domain/DomainBuilder.java [hunk 2]
```diff
@@ -79,4 +84,48 @@ private long id() {
...
+        var hexKey = Key.newBuilder().setEd25519(ByteString.copyFromUtf8(instr)).build().toByteArray();
+
+        Token.TokenBuilder builder = Token.builder()
+                .createdTimestamp(timestamp)
+                .decimals(1000)
+                .freezeDefault(false)
+                .freezeKey(hexKey)
+                .initialSupply(1_000_000L)
+                .kycKey(hexKey)
+                .maxSupply(1_000_000_000L)
```
Comment (ijungmann) [inline hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/domain/DomainBuilder.java:115]: nit: this doesn't line up with an `INFINITE` supply type.

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/repository/upsert/AbstractUpsertQueryGenerator.java [hunk 3]
```diff
@@ -94,7 +94,7 @@ protected boolean needsOnConflictAction() {
 
     @Override
     public String getCreateTempTableQuery() {
-        return String.format("create temporary table if not exists %s on commit drop as table %s limit 0",
+        return String.format("create temporary table if not exists %s as table %s limit 0",
```
Comment (ijungmann) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/repository/upsert/AbstractUpsertQueryGenerator.java:97]: I'm not entirely clear what effect this change has on the DB.

Comment (Nana-EC) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/repository/upsert/AbstractUpsertQueryGenerator.java:97]: Yeah, I'll add an item to the description comments. With the multiple connections the temp table needs to stay present across connections

**Comment by steven-sheehy** [comment]
Closing as not needed for now. Will keep branch.

**Pull request closed without merging**